docs/contribution_guide.md

```diff
@@ -77,3 +77,20 @@ https://captum.ai/tutorials/House_Prices_Regression_Interpret
 **Multimodal**
 - You can use VQA model and dataset described here:
 https://captum.ai/tutorials/Multimodal_VQA_Captum_Insights
+
+
+## Docstring style
+
+Docstring is required for all public APIs to provide users the details of the arguments and returns. [Our API documentation](https://captum.ai/api/) is generated from the docstring. Captum adopts a customized docstring format modified on top of [Google style](https://www.sphinx-doc.org/en/master/usage/extensions/example_google.html). Specifically, each argument should be listed as `arg_name (type): description` in the `Args:` section. The argument typing convention:
+- primitive types: `int`, `str`, `float`, `bool`
+- common collection types: `list`, `tuple`, `dict`
+    - [PEP 585](https://peps.python.org/pep-0585/#implementation) has deprecated the duplicate types: `List`, `Tuple`, `Dict`
+    - element types: `list[int]`, `dict[int, str]`
+- other foundamental types: `Any`, `Callable`, `Interable`
+- class types: `MyClass`, `external_lib.SomeClass`
+- omit `torch` for common Pytorch types: `Tensor`, `nn.Module`
+- use `or` and `,` for union types: `type1 or type2`, `type1, tyep2, or type3`
+    - [PEP 604](https://peps.python.org/pep-0604/) proposes to use `|` to connect types: `type1 | type2`. We may consider migration later.
```

Comment:
I don't think that Sphinx supports using "`|`" to separate types, so using it will break the hyperlinking. Up until recently it only supported the using "`or`", but it now also supports "`of`" as well.

Reply:
I would recommend just using "`or`" to connect types for now.

Reply:
again, this pr is to standardize docstring. For whatever is broken, we can fix it next, especially hyperlinks which never work as expected anyway.

Moreover, for this case, "`|`" is the recommended way in python typehints. I doubt sphinx does not support this delimiter. It should support all official typehints. You may try it with Captum if curious. But anyway, we are not migrating to it now.

Reply:
Sphinx's Python domain supports it according to the deliminator regex and supporting tests: https://github.com/sphinx-doc/sphinx/blob/5.x/sphinx/domains/python.py#L383. However, I was concerned about the napoleon extension's lack of explicit regex for detecting "`|`": https://github.com/sphinx-doc/sphinx/blob/5.x/sphinx/ext/napoleon/docstring.py

However, I just tested with Sphinx 4.3.2 and it appears to work:

![formatting_test1](https://user-images.githubusercontent.com/10626398/187723039-934cf9ea-bf6c-4215-994a-b2706dd297c9.JPG)

Sphinx's napoleon extension is really sparse on testing, so I suppose I should write some more tests in order to protect the usage of "`|`" from being broken.